### PR TITLE
Add objects to ResizeObserver IDL test

### PR DESCRIPTION
**Pull request by lukebjerring**
Adds a `ResizeObserverEntry` to test, obtained using the ResizeTestHelper resource.

### CODE REVIEW

File: resize-observer/idlharness.window.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+// https://wicg.github.io/ResizeObserver/
+idl_test(
+  ['ResizeObserver'],
+  ['dom', 'geometry'],
+  idl_array => {
+    try {
+      self.observer = new ResizeObserver(entries => {});
+      self.entry = new ResizeObserverEntry(document);
+      self.observation = new ResizeObservation(document);
```
Comment (atotic) [inline resize-observer/idlharness.window.js]: ResizeObservation is not supposed to be visible in the browser according to the spec.

Comment (lukebjerring) [reply resize-observer/idlharness.window.js]: Oh - I was under the impression the default exposure set was `Window`, and I'm pretty sure that assumption exists in [idlharness](https://github.com/web-platform-tests/wpt/blob/master/resources/idlharness.js#L681):

> let result = default_set || ["Window"];

That doesn't seem to be the case in https://heycam.github.io/webidl/#Exposed though.
Note that a bunch of tests (11, to be precise) fail for precisely that reason:

> assert_own_property: self does not have own property "ResizeObservation" expected property "ResizeObservation" missing

Comment (lukebjerring) [reply resize-observer/idlharness.window.js]: Found where I got that impression from:
From https://www.w3.org/TR/WebIDL-1/#Exposed

> If the [Exposed] extended attribute does not appear on a construct, then its exposure set is defined implicitly, depending on the type of construct:
> 
> interface
> dictionary
> The exposure set of the interface or dictionary only contains the primary global interface.

Comment (lukebjerring) [reply resize-observer/idlharness.window.js]: @heycam - could you clarify what expected behaviour here is? Has defaulting to `Window` changed from WebIDL v1?

File: interfaces/ResizeObserver.idl [hunk 2]
```diff
@@ -11,7 +11,7 @@ interface ResizeObserver {
     void disconnect();
 };
 
-callback ResizeObserverCallback = void (sequence<ResizeObserverEntry> entries, ResizeObserver observer);
+callback ResizeObserverCallback = void (sequence<ResizeObserverEntry> entries, ResizeObserver observer)
```
Comment (foolip) [inline interfaces/ResizeObserver.idl]: Where did the semicolon go?

File: resize-observer/idlharness.window.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    "ResizeObserverEntry creator",
+    [
+      {
+        setup: observer => {
+          self.observer = observer;
+          observer.observe(div);
+          div.style.width = "5px";
+        },
+        notify: entries => {
+          self.entry = entries[0];
```
Comment (foolip) [inline resize-observer/idlharness.window.js]: Turn this stuff into async setup now?

**Review by foolip — CHANGES_REQUESTED** [review]

File: interfaces/ResizeObserver.idl (hunk 2 above)
Comment (lukebjerring) [reply interfaces/ResizeObserver.idl]: It's been updated in the IDL (https://github.com/WICG/ResizeObserver/pull/56) but not pushed, I think.

File: resize-observer/idlharness.window.js (hunk 3 above)
Comment (lukebjerring) [reply resize-observer/idlharness.window.js]: Done.

**Review by foolip — APPROVED** [review]

**Pull request merged**